Xplat/src/main/java/vazkii/botania/common/entity/ManaSparkEntity.java

```diff
@@ -212,8 +223,35 @@ public void tick() {
 			}
 		}
 
-		if (removeTransferants > 0) {
-			removeTransferants--;
+		firstTick = false;
+	}
+
+	@Override
+	public void updateTransfers() {
+		switch (getUpgrade()) {
+			case RECESSIVE -> {
+				var otherSparks = SparkHelper.getSparksAround(level(), getX(), getY() + (getBbHeight() / 2), getZ(), getNetwork());
+				Collections.shuffle(otherSparks);
+				for (var otherSpark : otherSparks) {
+					SparkUpgradeType otherUpgrade = otherSpark.getUpgrade();
+					if (otherSpark != this
+							&& otherUpgrade != SparkUpgradeType.DOMINANT
+							&& otherUpgrade != SparkUpgradeType.RECESSIVE
+							&& otherUpgrade != SparkUpgradeType.ISOLATED) {
+						transfers.add(otherSpark);
+					}
+				}
+			}
+			case DOMINANT -> {
+				List<ManaSpark> validSparks = SparkHelper.getSparksAround(level(), getX(), getY() + (getBbHeight() / 2), getZ(), getNetwork());
+				for (var spark : validSparks) {
+					SparkUpgradeType otherUpgrade = spark.getUpgrade();
+					if (spark != this && otherUpgrade == SparkUpgradeType.NONE && spark.getAttachedManaReceiver() instanceof ManaPool) {
+						transfersTowardsSelfToRegister.add(spark);
+					}
+				}
+				Collections.shuffle(transfersTowardsSelfToRegister);
```

Comment:
reasoning for the shuffles that were added here?

Reply:
Otherwise mana pools would be loaded in a relatively consistent order, making only some fill up. This makes it so if they're reloaded, that order is shuffled at least sometimes. A similar thing was in here before,  I'm only guessing the reason why it has been implemented: https://github.com/VazkiiMods/Botania/pull/4459/files#diff-6ab7cbbb33c3d206fc8ca670958f924a1353804a7fb8f234f3308622d23723f7L172 
I can also add the transfers directly, but that would be slightly breaking with older behavior (Previosly, and in this implementation dominant sparks would only add one transfer each tick, taking some time to initialize). This would make `transfersTowardsSelfToRegister` unnecessary.